core/src/test/java/org/apache/calcite/util/format/FormatElementEnumTest.java

```diff
@@ -33,4 +33,130 @@ class FormatElementEnumTest {
     FormatElementEnum.DAY.format(ts, Date.from(Instant.parse("2014-09-30T10:00:00Z")));
     assertThat(ts, hasToString("Tuesday"));
   }
+
+  @Test void testD() {
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.D.format(ts, Date.from(Instant.parse("2014-09-30T10:00:00Z")));
```

Comment:
Since the date is the same for all tests maybe you can factor it out in a separate helper function.
This would also enable extending this testing code for other dates.

Reply:
I'm trying, and I seem to have found some problems. If there are too many problems, I will open a Jira and try to fix them in this PR.